src/praisonai/praisonai/integrations/compute/tenki.py

```
@@ -0,0 +1,312 @@
+        self._project_id = project_id or os.environ.get("TENKI_PROJECT_ID", "")
+        self._client = None
+        self._sandboxes: Dict[str, Dict[str, Any]] = {}
+
+    def _get_client(self):
+        if self._client is None:
+            try:
+                from tenki_sandbox import Client
+            except ImportError:
+                raise ImportError(
+                    "Tenki SDK required. Install with: pip install tenki-sandbox"
+                )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)